daphne_worker/src/dap.rs

```diff
@@ -274,38 +279,22 @@ impl DapReportInitializer for DaphneWorker<'_> {
         // Send ReportsProcessed requests.
         let mut reports_processed_requests = Vec::new();
         for (durable_name, consumed_reports) in reports_processed_request_data.into_iter() {
-            reports_processed_requests.push(durable.post_with_handler(
+            reports_processed_requests.push(durable.post(
                 BINDING_DAP_REPORTS_PROCESSED,
                 DURABLE_REPORTS_PROCESSED_INITIALIZE,
                 durable_name,
                 consumed_reports,
-                |output: ReportsProcessedResp, retried| (output, retried),
             ));
         }
-        let reports_processed_responses: Vec<(ReportsProcessedResp, bool)> =
+        let reports_processed_responses: Vec<ReportsProcessedResp> =
             try_join_all(reports_processed_requests)
                 .await
                 .map_err(|e| fatal_error!(err = e))?;
 
         // Flatten the responses from ReportsProcessed into a hash map.
         let mut initialized_reports = HashMap::new();
-        for (reports_processed_response, retried) in reports_processed_responses.into_iter() {
-            for mut initialized_report in reports_processed_response.initialized_reports.into_iter()
-            {
-                if let EarlyReportStateInitialized::Rejected {
-                    ref mut failure, ..
-                } = initialized_report
-                {
-                    // The request to ReportsProcessed is not idempotent. If it was retried, then
-                    // it's possible that one or more of the reports were erroneously marked as
-                    // replayed. This would happen if, for example, a fresh report was marked as
-                    // aggregated in the failed request, then marked as replayed by a retried
-                    // request. In this case we cannot determine definitively if the report was
-                    // replayed, so we drop it to be safe.
```

Comment:
We no longer mark the reports as aggregated, so this logic for handling failed DO requests is no longer necessary. Retry should be safe.